pkg/labels/labels.go

```diff
@@ -0,0 +1,34 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package labels
+
+import "sigs.k8s.io/controller-runtime/pkg/client"
+
+// Exist returns true if obj contains labels m.
+func Exist(obj client.Object, m map[string]string) bool {
+	labels := obj.GetLabels()
+	if labels == nil {
+		return false
+	}
+	desired := len(m)
+	found := 0
+	for l, v := range labels {
+		for k, w := range m {
+			if l == k && v == w {
+				found++
+			}
+		}
+	}
+	return found == desired
```

Comment:
this would be more efficient&simpler using map lookups like so:

```go
for key, val := range m {
    if found, ok := labels[key]; !ok || found != val {
        return false
    }
}
return true
```

Reply:
@skriss thanks for the pointer. Commit `8e778bc` resolves this comment.